src/tpm2_wrap.c

```diff
@@ -2897,6 +2897,70 @@ int wolfTPM2_HmacFinish(WOLFTPM2_DEV* dev, WOLFTPM2_HMAC* hmac,
     return rc;
 }
 
+/* performs a reset sequence */
+int wolfTPM2_Shutdown(WOLFTPM2_DEV* dev, int doStartup)
+{
+    int rc;
+    Shutdown_In shutdownIn;
+    Startup_In startupIn;
+
+    if (dev == NULL) {
+        return BAD_FUNC_ARG;
+    }
+
+    /* shutdown */
+    XMEMSET(&shutdownIn, 0, sizeof(shutdownIn));
+    shutdownIn.shutdownType = TPM_SU_CLEAR;
+    rc = TPM2_Shutdown(&shutdownIn);
+    if (rc != TPM_RC_SUCCESS) {
+    #ifdef DEBUG_WOLFTPM
+        printf("TPM2_Shutdown failed 0x%x: %s\n", rc, TPM2_GetRCString(rc));
+    #endif
+    }
+
+    /* startup */
+    if (doStartup) {
```

Comment:
Should the startup continue even if the shutdown failed?

Reply:
Yes, I'd like it to try and startup even if there was a shutdown failure... Assuming `doStartup` is set.